src/shims/unix/socket.rs

```diff
@@ -63,6 +111,16 @@ impl FileDescription for SocketPair {
             return Ok(Ok(0));
         }
 
+        // In libc socketpair, reading from a previously full socketpair buffer won't trigger
+        // epoll notification. We want to inform the user about this inconsistency in libc and
+        // hope to receive a bug report if someone has a use case for this.
+        if readbuf.buf.len() == MAX_SOCKETPAIR_BUFFER_CAPACITY {
+            throw_unsup_format!(
+                "A previously full socketpair has been read, please submit a \
+            bug report in https://github.com/rust-lang/miri/issues/new"
+            );
+        }
+
```

Comment:
@RalfJung We observed that in libc, reading from a previously full socketpair buffer doesn't trigger any epoll notification. This behavior seems buggy because reading from a full buffer allows a previously blocked ``write`` to succeed. So we want a mechanism to notify the user to submit a bug report to us. We can either ``throw_unsup_format`` or emit a warning then let miri continue executing. Do you have any preference on how we do this? 


cc @oli-obk

Reply:
That sounds like very strange behavior -- do you have a self-contained example? Have you brought this up with an epoll expert to get an explanation?

"Notify me when I can write to this buffer again" is an intended epoll usecase, AFAIK.